src/NUnitFramework/framework/Internal/Randomizer.cs

```diff
@@ -82,15 +107,12 @@ public static Randomizer GetRandomizer(ParameterInfo parameter)
         /// <summary>
         /// Create a new Randomizer using the next seed
         /// available to ensure that each randomizer gives
-        /// a unique sequence of _values.
+        /// a unique sequence of values.
         /// </summary>
         /// <returns></returns>
         public static Randomizer CreateRandomizer()
         {
-            if (seedGenerator == null)
-                seedGenerator = new Random(InitialSeed);
-
-            return new Randomizer(seedGenerator.Next());
+            return new Randomizer(SeedGenerator.Next());
```

Comment:
Further to the thread safety, if tests are run in parallel, we cannot guarantee the order they are run in. I assume that these are created as we are creating the tests, not running, so they are in a repeatable order?